File: C++Programming/polymorphism/polymorphism/SalariedEmployee.cpp
//
//  SalariedEmployee.cpp
//  polymorphism
//
//  Created by Vijay Kumar on 9/18/13.
//  Copyright (c) 2013 Vijay Kumar. All rights reserved.
//

#include "SalariedEmployee.h"
#include <iostream>
#include <string>

using namespace std;

SalariedEmployee::SalariedEmployee(const string& first_name, const string& last_name,
                                   const string& ssn, const double salary_passed)
:Employee(first_name, last_name, ssn), salary(salary_passed)
{
  
}

void SalariedEmployee::print() const
{
  cout << "Salaried Employee: " << endl;
  Employee::print();
}


double SalariedEmployee::earnings() const
{
  return salary;
}

File: LearningJava/Polymorphism/SalariedEmployee.java
public class SalariedEmployee extends Employee
{
  private double weeklySalary;
  
  public SalariedEmployee(String firstName, String lastName, String socialSecutiryNumber, double weeklySalary)
  {
    super(firstName, lastName, socialSecutiryNumber);
    
    if(weeklySalary < 0)
      throw new IllegalArgumentException("Weekly salary must be > 0.");
    
    this.weeklySalary = weeklySalary;
  }
  
  @Override
  public double earnings()
  {
    return weeklySalary;
  }
  
  @Override
  public String toString()
  {
    return String.format("%sweeklySalary = %.2f%n", super.toString(), weeklySalary);
  }
}